pandas/core/frame.py

```diff
@@ -7182,13 +7182,16 @@ def sort_values(
         elif len(by):
             # len(by) == 1
 
-            k = self._get_label_or_level_values(by[0], axis=axis)
+            if by[0] is None and None in self.columns and axis == 0:
+                k = self[None]._values
```

Comment:
This seems to workaround a limitation of `_get_label_or_level_values` which we should avoid